### PR TITLE
Move to pip-compile-multi for easier deps management

### LINKED ISSUE #603: Move dependency management to pip-compile-multi
### Description
As part of standarising how we manage our Python deps across MEAO, we're moving to [pip-compile-multi](https://pip-compile-multi.readthedocs.io/en/latest/)

### Key tasks

- [ ] Revamp dependency management tooling from `pip` to `pip-compile-multi`
- [ ] Expand Docker setup to allow for a two builds: Dev/`test`/local-dev build, and a `release` build, the same as with Bedrock

### Success critera
- [ ] CI passes
- [ ] No regression in behaviour/functionality
- [ ] Documentation is updated
- [ ] Separate documentation site can still be built 
- [ ] Project can be installed locally with no impediments

### PR DESCRIPTION
**Pull request by robhudson**
Fixes #603

### CODE REVIEW

**Comment by robhudson** [comment]
From the test run I'm seeing this warning:
`RequestsDependencyWarning: urllib3 (1.26.6) or chardet (3.0.4) doesn't match a supported version!`

File: bin/compile-requirements.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+set -exo pipefail
+
+export CUSTOM_COMPILE_COMMAND="make compile-requirements"
+
+# We need this installed, but we don't want it to live in the main requirements
+# We will need to periodically review this pinning
+pip install --upgrade "pip<22"  # Temporary until pip-tools is fixed.
```
Comment (stevejalim) [inline bin/compile-requirements.sh:9]: Nice way to handle it

File: bin/compile-requirements.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+pip install --upgrade pip-tools==6.4.0  # Needs at least this version to build.
+pip install --upgrade pip-compile-multi
+
+pip-compile-multi --generate-hashes prod --generate-hashes dev
```
Comment (stevejalim) [inline bin/compile-requirements.sh:13]: Suggestionm, if you want: Bedrock now has an additional param that lets us add a custom header to explain how to rebuild deps, replacing the default which is misleading in this context

`--header=/app/bin/pip-compile-multi-header-message.txt`

File: requirements/dev.txt [hunk 3]
```diff
@@ -0,0 +1,105 @@
+# SHA1:0e3b4eb4275a8a7777e75467fd7ca031a2d61a9c
+#
+# This file is autogenerated by pip-compile-multi
+# To update, run:
+#
+#    pip-compile-multi
```
Comment (stevejalim) [inline requirements/dev.txt:6]: This is the message we could override with a custom header, if you want

**Review by stevejalim — APPROVED** [review]
Nice one. r+wc

It built fine and passed tests for me locally on Apple Silicon, plus running `make compile-requirements` operated fine, picking up a bump from Pillow 9.0.0 to 9.0.1 (which might be worth folding in here if it's a bugfix release)

**Pull request merged**